src/Compilers/CSharp/Portable/Symbols/Source/SourceExtensionTypeSymbol.cs

```diff
@@ -61,6 +61,17 @@ protected override void CheckUnderlyingType(BindingDiagnosticBag diagnostics)
                 var location = singleDeclaration.NameLocation;
 
                 underlyingType.CheckAllConstraints(DeclaringCompilation, conversions, location, diagnostics);
+
+                if (!IsExplicitExtension)
+                {
+                    foreach (var typeParameter in TypeParameters)
```

Comment:
>foreach (var typeParameter in TypeParameters) [](http://example.com/codeflow?start=20&length=45)

Consider using ```TypeSymbolExtensions.VisitType``` to visit ```underlyingType``` once and collecting all type parameters that it references in a HashSet. Then doing lookup in that HashSet in a loop.